src/VisualStudio/Core/Def/Implementation/TableDataSource/VisualStudioDiagnosticListTable.cs

```diff
@@ -26,6 +26,7 @@ internal partial class VisualStudioDiagnosticListTable : VisualStudioBaseDiagnos
 
         [ImportingConstructor]
         public VisualStudioDiagnosticListTable(
+            IVsErrorList errorList,
```

Comment:
Not all types available through `IServiceProvider` are available through MEF. This is one of those types. Since you can't make the constructor for `VisualStudioDiagnosticListTable` asynchronous, there are three primary options here:

1. Keep the constructor here the same as it was before, but ensure `SVsErrorList` has already been requested. In other words, update `LoadComponentsInUIContextAsync` to call `GetServiceAsync(typeof(SVsErrorList))` before requesting an instance of this MEF type. This will ensure the interface is readily available when you `GetService` on the UI thread.
2. Switch from simple construction to a construction+async initialization approach. After requesting the instance of `VisualStudioDiagnosticListTable`, immediately `await InitializeAsync`, and request the instance of `SVsErrorList` there.
3. Make `SVsErrorList` lazily requested by asynchronous methods in the type, so the constructor is just a simple non-blocking constructor.

Option 1 is the easiest, but it's also fragile. I have a feeling there is a good fourth option, where we can use MEF metadata to declare the dependencies and then initialize them.

```csharp
// placed on the VisualStudioDiagnosticListTable class
[DependsOnService(typeof(SVsErrorList))]

// Pseudo-code for LoadComponentsInUIContextAsync
foreach (var requiredService in GetRequiredServices())
{
  await GetServiceAsync(requiredService);
}
```